pkg/subctl/cmd/join.go

```diff
@@ -279,13 +279,16 @@ func joinSubmarinerCluster(subctlData *datafile.SubctlData) {
 	utils.ExitOnError("Error creating SA for cluster", err)
 
 	// We need to connect to the broker in all cases
-	brokerSecret, err := brokersecret.Ensure(clientConfig, OperatorNamespace, populateBrokerSecret(subctlData))
+	brokerSecret, err := secret.Ensure(clientConfig, OperatorNamespace, populateBrokerSecret(subctlData))
 	utils.ExitOnError("Error creating broker secret for cluster", err)
 
 	if subctlData.IsConnectivityEnabled() {
 		status.Start("Deploying Submariner")
 
-		err = submarinercr.Ensure(clientConfig, OperatorNamespace, populateSubmarinerSpec(subctlData, brokerSecret, netconfig))
+		pskSecret, err := secret.Ensure(clientConfig, OperatorNamespace, populatePSKSecret(subctlData))
+		utils.ExitOnError("Error creating PSK secret for cluster", err)
```

Comment:
This will also need to be integrated into the relocated **join** code with https://github.com/submariner-io/submariner-operator/pull/1726.